Add Model classes

viacheslaev opened the pull request.

viacheslaev committed:
Add Model classes

stvort commented on src/main/java/ru/otus/rzdtelegrambot/model/Train.java:
```diff
@@ -0,0 +1,44 @@
+    //код станции отправления
+    private String route0;
+    //код станции прибытия
+    private String route1;
```
Лучше все же sourceSatation/destinationStation, startPoint/destinationPoint

stvort commented on src/main/java/ru/otus/rzdtelegrambot/model/Train.java:
```diff
@@ -0,0 +1,44 @@
+    //дата прибытия
+    private Date date1;
+    //время прибытия
+    private Date time1;
```
Тоже самое. Если нужно писать комментарии значит что-то не так с названиями

stvort commented on src/main/java/ru/otus/rzdtelegrambot/model/CarType.java:
```diff
@@ -0,0 +1,20 @@
+public enum CarType {
+    PLACKART("Плацкарт"),
+    KUPE("Купе"),
+    SV("СВ");
```
У всех этих слов есть перевод на английский. Ну и я бы не хардкодил русские слова. У нас же вроде бут? Посмотрите как можно работать с MessageSource в нем

viacheslaev committed:
Refactor: add Locale Messages for CarTypes enum.

viacheslaev committed:
Add CarTypeLocalization Service

viacheslaev committed:
Refactor Bot config

viacheslaev committed:
Add simple keyboard methods

viacheslaev committed:
Refactor to WebHookBot

viacheslaev committed:
Add Users message parsing (dialog) for train search.

viacheslaev committed:
Add TrainSearch saves Users State

viacheslaev committed:
Refactor: add MessageHandlers

viacheslaev committed:
Refactor: realize BotState pattern

viacheslaev committed:
Renamed MessageHandler

viacheslaev committed:
Add: services

stvort commented on src/main/resources/application.properties:
```diff
@@ -1,12 +1,10 @@
+#spring.main.web-application-type=none
```
А попробуйте yml формат для свойств. Он для бута чаще используется. Не относится к бандлам

stvort commented on src/main/resources/application.properties:
```diff
@@ -1,12 +1,10 @@
-telegrambot.proxyType=SOCKS5
-telegrambot.proxyHost=localhost
-telegrambot.proxyPort=9150
-
```
А зачем разделили?

stvort commented on src/main/java/ru/otus/rzdtelegrambot/model/TrainTicketsInfoSubscription.java:
```diff
@@ -0,0 +1,25 @@
+    private Date dateDepart;
+
+    //Тип вагона и минимальная цена на билет
+    Map<CarType, Integer> carsTariffs;
```
Лучше сделать отдельным классом. Или не делать) Но главное саму мапу наружу не выставлять. Через методы с ней работать

stvort commented on src/main/java/ru/otus/rzdtelegrambot/botapi/handlers/trainsearch/TrainSearchHandler.java:
```diff
@@ -0,0 +1,128 @@
+    private BotStateContext botStateContext;
+    private TrainTicketsInfoService trainTicketsService;
+    private StationCodeService stationCodeService;
+    private RZDTelegramBot telegramBot;
```
Все что пришло в конструктор может быть final

stvort commented on src/main/java/ru/otus/rzdtelegrambot/botapi/handlers/trainsearch/TrainSearchHandler.java:
```diff
@@ -0,0 +1,128 @@
+        return BotState.TRAINS_SEARCH;
+    }
+
+    private SendMessage processUsersInput(Message inputMsg) {
```
Огромный метод. Можно разбить скорее всего

stvort commented on src/main/java/ru/otus/rzdtelegrambot/botapi/handlers/trainsearch/TrainSearchRequestData.java:
```diff
@@ -0,0 +1,29 @@
+public class TrainSearchRequestData {
+    String departureStation;
+    String arrivalStation;
+    Date dateDepart;
```
Ну ИМХО лучше написать всем private) Так или иначе лучше придерживаться одного стиля

stvort commented on src/main/java/ru/otus/rzdtelegrambot/botconfig/RZDTelegramBotConfig.java:
```diff
@@ -1,25 +1,52 @@
+@Configuration
+@PropertySource("classpath:telegrambot.properties")
 @ConfigurationProperties(prefix = "telegrambot")
 @Getter
 @Setter
+@FieldDefaults(level = AccessLevel.PRIVATE)
 public class RZDTelegramBotConfig {
```
Не уверен, что конфигу нужны геттеры и сеттеры

stvort commented on src/main/java/ru/otus/rzdtelegrambot/service/CarTypeLocalizationService.java:
```diff
@@ -0,0 +1,57 @@
+        ECONOMY_SLEEPING = resourceBundle.getString("ECONOMY_CLASS_SLEEPING");
+        FIRST_SLEEPING = resourceBundle.getString("FIRST_CLASS_SLEEPING");
+        SECOND_SLEEPING = resourceBundle.getString("SECOND_CLASS_SLEEPING");
+    }
```
В буте обычно все же используют `MessageSource`. Его можно внедрять. Ключи в бандле пишутся как обычные настройки. Маленькими буквами с разделителем точкой

stvort commented on src/main/java/ru/otus/rzdtelegrambot/service/CarTypeLocalizationService.java:
```diff
@@ -0,0 +1,57 @@
+    private String ECONOMY_SITTING;
+    private String ECONOMY_SLEEPING;
+    private String FIRST_SLEEPING;
+    private String SECOND_SLEEPING;
```
Это не константы. Обычные переменные. Так что lowerCamelCase)

stvort commented on src/main/java/ru/otus/rzdtelegrambot/service/CarTypeLocalizationService.java:
```diff
@@ -0,0 +1,57 @@
+            case SECOND_CLASS_SLEEPING:
+                localizedCarType = SECOND_SLEEPING;
+                break;
+        }
```
А что если в мапу сложить все это дело? `Map<CarType, String>`? Или просто доставать из бандла сразу (можно закешировать).

stvort commented on src/main/java/ru/otus/rzdtelegrambot/service/MainMenuService.java:
```diff
@@ -0,0 +1,56 @@
+ * @author Sergei Viacheslaev
+ */
+@Component
+public class MainMenuService {
```
Если оно Service, то почему `@Component`?

stvort commented on src/main/java/ru/otus/rzdtelegrambot/service/MainMenuService.java:
```diff
@@ -0,0 +1,56 @@
+
+        KeyboardRow row = new KeyboardRow();
+        row.add(new KeyboardButton("Найти поезда"));
+        row.add(new KeyboardButton("Помощь"));
```
Магические строки и числа лучше сделать константами. Ну и их не локализуем?

stvort commented on src/main/java/ru/otus/rzdtelegrambot/service/StationCodeService.java:
```diff
@@ -0,0 +1,64 @@
+@FieldDefaults(level = AccessLevel.PRIVATE)
+public class StationCodeService {
+    final String stationCodeRequestTemplate = "https://pass.rzd.ru/suggester?stationNamePart={stationNamePart}&lang=ru";
+    final RestTemplate restTemplate = new RestTemplate();
```
Зависимости лучше извне получать. RestTemplate можно сделать бином. А то как тестировать потом)

stvort commented on src/main/java/ru/otus/rzdtelegrambot/service/StationCodeService.java:
```diff
@@ -0,0 +1,64 @@
+        if (stationCode != null) {
+            return stationCode;
+        } else {
+            throw new RuntimeException("ERROR ! stationCode= null");
```
По идее можно прямо свое исключение сделать. `IvalidStationCodeException`

stvort commented on src/main/java/ru/otus/rzdtelegrambot/service/TrainTicketsInfoService.java:
```diff
@@ -0,0 +1,141 @@
+public class TrainTicketsInfoService {
+    private final String trainInfoRidRequest = "https://pass.rzd.ru/timetable/public/ru?layer_id=5827&dir=0&tfl=3&" +
+            "checkSeats=1&code0={STATION_DEPART_CODE}&dt0={DATE_DEPART}&code1={STATION_ARRIVAL_CODE}";
+    private String trainInfoDataRequestTemplate = "https://pass.rzd.ru/timetable/public/ru?layer_id=5827&rid={RID_VALUE}";
```
Это вот как раз может быть константами

stvort commented on src/main/java/ru/otus/rzdtelegrambot/service/TrainTicketsInfoService.java:
```diff
@@ -0,0 +1,141 @@
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    private HttpHeaders httpHeaders;
+    private String rid;
```
Не уверен, что сервису нужно такое состояние. Больше похоже на то что это может быть локальными переменными

stvort commented on src/main/java/ru/otus/rzdtelegrambot/service/TrainTicketsInfoService.java:
```diff
@@ -0,0 +1,141 @@
+    private String rid;
+    private final SimpleDateFormat dateFormatter = new SimpleDateFormat("dd.MM.yyyy");
+
+    private final RestTemplate restTemplate = new RestTemplate();
```
Вот) Точно его бином надо делать

stvort commented on src/main/java/ru/otus/rzdtelegrambot/service/TrainTicketsInfoService.java:
```diff
@@ -0,0 +1,141 @@
+        String brand;
+        String stationDepart;
+        String stationArrival;
+        String timeDepart;
```
Вроде не используется)

viacheslaev committed:
Add: Emoji Support

viacheslaev committed:
Add: button Subscribe support

viacheslaev committed:
Add: parse CallBackQuery Data

viacheslaev committed:
Add: mongoDB embedded

viacheslaev committed:
Refactor: prepare for heroku

viacheslaev committed:
Refactor: add gradle-wrapper.jar

viacheslaev committed:
Add: logic "Unsubscribe"

viacheslaev committed:
Refactor: for Heroku

viacheslaev committed:
Refactor: add getCarsWithMinimalPrice method

viacheslaev committed:
Refactor: replace UserDataCache to WeakHashMap

viacheslaev committed:
Refactor: add notificationService

viacheslaev committed:
Add: pinging Task for Herocu

viacheslaev committed:
Refactor: SubscriptionsMenuHandler

viacheslaev committed:
Refactor: buttons will change status, if pressed.

viacheslaev committed:
Refactor: fix subscribe bugs

viacheslaev committed:
Add: CarsProccessingService

viacheslaev committed:
Refactor: notification  price change (processCarsList)

viacheslaev committed:
Refactor: optimize TicketsNotificationService, now sends one message-notification.

viacheslaev committed:
Refactor: add CallbackQuery handlers.

viacheslaev committed:
Fix: dateDepart out of bound 90 days error.

viacheslaev committed:
Fix: price change notification message.

viacheslaev committed:
Refactor: rename method

viacheslaev committed:
Refactor: fix botstatecontext

stvort commented on src/main/java/ru/otus/rzdtelegrambot/botapi/handlers/callbackquery/CallbackQueryFacade.java:
```diff
@@ -0,0 +1,60 @@
+            default:
+                queryReply = new SendMessage(callbackQuery.getMessage().getChatId(), NotificationMessage.HANDLE_KEYBOARD_QUERY_FAILED.toString());
+                break;
+        }
```
`case` почти всегда является звонком, что что-то не так в архитектурном плане. Предлагаю всю обработку сделать хендлерами. Внутри хендлера сделать поле, отвечающее за то какой `queryType` он может обрабатывать. А сами хендлеры сложить в список. Спринг может автовайрить в список все бины одного типа

viacheslaev replied:
Согласен, сделал, как вы рекомендовали. Так действительно короче получается и понятнее.

stvort commented on src/main/java/ru/otus/rzdtelegrambot/botapi/handlers/callbackquery/SubscribeTicketsInfoQueryHandler.java:
```diff
@@ -0,0 +1,80 @@
+
+        return new SendMessage(callbackQuery.getMessage().getChatId(),
+                String.format("Оформлена подписка на поезд №%s отправлением %s",
+                        carsProcessingService.parseTrainNumber(callbackQuery), carsProcessingService.parseDateDepart(callbackQuery)));
```
Тут и далее, я все же советую не держать тест внутри кода. Давайте хотя бы сосредоточим его в одном-двух классах в виде констант/енумов. Это тоже так себе вариант, но потенциально позволит в дальнейшем не разыскивать все места с текстом чтобы локализовать все это или просто поменять. В идеале у нас д.б. сервис сообщений, который работает с бандлами через MessageSource. Т.е. в данном, конкретном случае тут могло бы быть:
```
SendMessage(callbackQuery.getMessage().getChatId(), messageService.getMessage("train.subscription.created", 
                        carsProcessingService.parseTrainNumber(callbackQuery), 
                        carsProcessingService.parseDateDepart(callbackQuery)));
```

viacheslaev replied:
Создал сервис, который используя файл messages_ru_RU и MessageSource, формирует готовое ответное сообщение.

stvort commented on src/main/java/ru/otus/rzdtelegrambot/botapi/handlers/callbackquery/SubscribeTicketsInfoQueryHandler.java:
```diff
@@ -0,0 +1,80 @@
+                callbackMessage.indexOf(",")).trim();
+        String stationArrival = callbackMessage.substring(callbackMessage.lastIndexOf("Прибытие:") + 10,
+                callbackMessage.lastIndexOf(",")).trim();
+        List<Car> availableCars = carsProcessingService.parseCarsFromMessage(callbackMessage);
```
Тут как бы тоже не совсем понятно что происходит) Но опять же лучше не завязываться на текст и уже тем более храдкодить его длину (сейчас если изменится строка, то размер надо будет менять отдельно). Обычно можно обойтись без этого.  Если я правильно понял то `callbackMessage` это сообщение под которым нажали "Подписаться". Если все так то разве нельзя в кнопку записать полный JSON со всем что нужно? Мне казалось примерно это и происходит в `SendTicketsInfoService#sendTrainTicketsInfo`
```
/Посылаем кнопку "Подписаться" с данными поезда на который подписываемся
            String callbackData = String.format("%s|%s|%s", CallbackQueryType.SUBSCRIBE,
                    train.getNumber(), train.getDateDepart());
```

viacheslaev replied:
Конечно- я так и хотел с самого начала и сделал, а потом начали валиться ошибки. Оказалось, что у Телеграма в объекте callbackQuery, что под кнопкой - максимальная длинна строки 64 байта) Поэтому приходиться часть данных выдерать из сообщения, а часть "из под кнопки callback"

stvort replied:
А если хранить в БД данные запросов а в callbackQuery только держать id? Просто выдирать из сообщения это только в качестве временной меры годится. Костыль. Ну и можно не все запросы хранить, а только N последних или N за последний час

viacheslaev replied:
Сделал через кэш, при каждом новом поиске сохраняется список поездов. А если пользователь захочет подписаться по старым предыдущим сообщениям, то ему напишут чтобы он повторил поиск. Убрал таким образом парсинг сообщения и работаю с объектами.

stvort commented on src/main/java/ru/otus/rzdtelegrambot/service/StationCodeService.java:
```diff
@@ -0,0 +1,68 @@
+@Slf4j
+@FieldDefaults(level = AccessLevel.PRIVATE)
+public class StationCodeService {
+    final String STATIONCODE_REQUEST_TEMPLATE = "https://pass.rzd.ru/suggester?stationNamePart={stationNamePart}&lang=ru";
```
`static final`

stvort commented on src/main/java/ru/otus/rzdtelegrambot/service/StationCodeService.java:
```diff
@@ -0,0 +1,68 @@
+        if (stationCode != null) return stationCode;
+
+        if (processStationCodeRequest(stationNameParam) != 0) {
+            return -1;
```
А вот тут енум бы зашел ИМХО

viacheslaev replied:
Иправил на Optional

stvort commented on src/main/java/ru/otus/rzdtelegrambot/service/TrainTicketsGetInfoService.java:
```diff
@@ -0,0 +1,150 @@
+public class TrainTicketsGetInfoService {
+    private final String TRAIN_INFO_RID_REQUEST = "https://pass.rzd.ru/timetable/public/ru?layer_id=5827&dir=0&tfl=3&" +
+            "checkSeats=1&code0={STATION_DEPART_CODE}&dt0={DATE_DEPART}&code1={STATION_ARRIVAL_CODE}";
+    private final String TRAIN_INFO_REQUEST_TEMPLATE = "https://pass.rzd.ru/timetable/public/ru?layer_id=5827&rid={RID_VALUE}";
```
`private static final`

stvort commented on src/main/java/ru/otus/rzdtelegrambot/service/TrainTicketsGetInfoService.java:
```diff
@@ -0,0 +1,150 @@
+        Map<String, String> urlParams = new HashMap<>();
+        urlParams.put("STATION_DEPART_CODE", String.valueOf(stationDepartCode));
+        urlParams.put("STATION_ARRIVAL_CODE", String.valueOf(stationArrivalCode));
+        urlParams.put("DATE_DEPART", dateDepartStr);
```
Магические строки лучше сделать константами

viacheslaev replied:
Сделал константами.

stvort commented on src/main/java/ru/otus/rzdtelegrambot/service/TrainTicketsGetInfoService.java:
```diff
@@ -0,0 +1,150 @@
+        urlParams.put("STATION_ARRIVAL_CODE", String.valueOf(stationArrivalCode));
+        urlParams.put("DATE_DEPART", dateDepartStr);
+
+        //1. Get RID and cookies
```
Это тоже сигнал) Метод огромный. Надо дробить

stvort commented on src/main/java/ru/otus/rzdtelegrambot/service/UserSubscriptionProcessService.java:
```diff
@@ -0,0 +1,169 @@
+            }
+        });
+
+        sleep(250);
```
А зачем? Лучше либо константу, которая все объяснит, либо в метод вынести

viacheslaev replied:
Сделал константу с понятным названием.

stvort commented on src/main/java/ru/otus/rzdtelegrambot/utils/PingTask.java:
```diff
@@ -0,0 +1,32 @@
+    @Scheduled(fixedRate = 900_000)
+    public void pingMe() {
+        try {
+            URL url = new URL("https://rzdbot-1035774480.herokuapp.com/");
```
В целом я бы вынес все урлы в application.properties

viacheslaev replied:
Вынес все url в application.properties

viacheslaev committed:
Refactor:
  1. CallbackQueryHandler: add fietd handlerQueryType,
     change switch to stream.filter() in processCallbackQuery

  2. Add ReplyMessageService - uses MessageSource

viacheslaev committed:
Refactor: subscribe/unsubscribe parse data and view.

viacheslaev committed:
Refactor: TrainTicketsGetInfoService

viacheslaev committed:
Refactor: view price changes at UserSubscriptionProcessService

viacheslaev committed:
Refactor: create StationsCache, add StationBook service and menu.

viacheslaev committed:
fix: only UserDataCache now stores Botstate.

stvort commented on src/main/java/ru/otus/rzdtelegrambot/botapi/handlers/callbackquery/CallbackQueryFacade.java:
```diff
@@ -0,0 +1,44 @@
+            userReply = queryHandler.get().handleCallbackQuery(usersQuery);
+        } else {
+            userReply = messagesService.getWarningReplyMessage(usersQuery.getMessage().getChatId(), "reply.query.failed");
+        }
```
Или так
```
        SendMessage userReply = queryHandler.map( h -> h.handleCallbackQuery(usersQuery))
                .orElse(messagesService.getWarningReplyMessage(usersQuery.getMessage().getChatId(), "reply.query.failed"));
```

viacheslaev replied:
Исправил, надо запомнить что с Optional тоже можно работать как со Stream.

stvort commented on src/main/java/ru/otus/rzdtelegrambot/service/TrainTicketsGetInfoService.java:
```diff
@@ -0,0 +1,193 @@
+    }
+
+    private boolean isResponseBodyHasNoTrains(String jsonRespBody) {
+        return jsonRespBody == null || jsonRespBody.contains("находится за пределами периода");
```
Тут тоже текст лучше не хардкодить)

stvort commented on src/main/java/ru/otus/rzdtelegrambot/service/UserSubscriptionProcessService.java:
```diff
@@ -0,0 +1,161 @@
+                if (actualCar.getCarType().equals(subscribedCar.getCarType())) {
+
+                    if (actualCar.getMinimalPrice() > subscribedCar.getMinimalPrice()) {
+                        notificationMessage.append(String.format("%sВозросла цена на вагоны %s, %s -> %s ₽.%n", Emojis.NOTIFICATION_PRICE_UP,
```
И тут)

stvort commented on src/main/java/ru/otus/rzdtelegrambot/botapi/BotStateContext.java:
```diff
@@ -0,0 +1,79 @@
+ */
+@Component
+public class BotStateContext {
+    private InputMessageHandler currentMessageHandler;
```
А зачем оно полем?

viacheslaev replied:
Убрал.

stvort commented on src/main/java/ru/otus/rzdtelegrambot/botconfig/RZDTelegramBotConfig.java:
```diff
@@ -1,25 +1,50 @@
  */
 
+@Configuration
 @ConfigurationProperties(prefix = "telegrambot")
```
Лучше сделать отдельно бин с настройками (`@ConfigurationProperties + @Component`) и отдельно конфиг (`@Configuration`), куда этот бин внедрить

viacheslaev replied:
Сделал отдельный бин с настройками и его внедрил в AppConfig

stvort commented on src/main/java/ru/otus/rzdtelegrambot/utils/PingTask.java:
```diff
@@ -0,0 +1,42 @@
+@Slf4j
+@Getter
+@Setter
+public class PingTask {
```
Наверное стоит его перенести из utils в service

viacheslaev replied:
Перенес.

stvort commented on src/main/java/ru/otus/rzdtelegrambot/service/ReplyMessagesService.java:
```diff
@@ -0,0 +1,76 @@
+    }
+
+    public SendMessage getTrainSubscribedMessage(long chatId, String trainNumber, String dateDepart) {
+        return new SendMessage(chatId, String.format(messageSource.getMessage("reply.query.train.subscribed", null, locale),
```
MessageSource умеет форматированные сообщения. ЕМНИП через `{0}/{n}`.Вот туда, где null передавать параметры.

stvort commented on src/main/java/ru/otus/rzdtelegrambot/service/ReplyMessagesService.java:
```diff
@@ -0,0 +1,76 @@
+    private final Locale locale;
+
+    private MessageSource messageSource;
+
```
Лучше еще в один сервис завернуть. Чтобы локаль и `MessageSource` вместе внутри и пара перегрузок `getMessage` (с параметрами и без, причем параметры можно у этого сервиса вараргами сделать). Вызов такого метода будет короче чем у стандартного от `MessageSource` за счет локали и параметров

viacheslaev replied:
Сделал отдельный сервис LocamMessageService с перегруженными методами.

viacheslaev committed:
Refactor: parseQueryData now uses UseDataCache, removed message parsing.

viacheslaev committed:
Refactor: add LocaleMessageService.

viacheslaev committed:
Refactor: reply messages

viacheslaev committed:
Refactor: view notifications.

viacheslaev committed:
Add auto-delete Subscription if train already left.

viacheslaev committed:
Add system.properties heroku, Java 12.

viacheslaev committed:
Fix: getStationDepart mistake

viacheslaev committed:
Refactor: add ParseQueryDataService

stvort commented on src/main/java/ru/otus/rzdtelegrambot/botapi/BotStateContext.java:
```diff
@@ -0,0 +1,75 @@
+    }
+
+    private boolean isTrainSearchState(BotState currentState) {
+        switch (currentState) {
+            case TRAINS_SEARCH:
+            case ASK_DATE_DEPART:
+            case DATE_DEPART_RECEIVED:
+            case ASK_STATION_ARRIVAL:
+            case ASK_STATION_DEPART:
+            case TRAINS_SEARCH_STARTED:
+            case TRAIN_INFO_RESPONCE_AWAITING:
+            case TRAINS_SEARCH_FINISH:
+                return true;
+            default:
+                return false;
+        }
```
В теории их можно в мапу сложить и передать сюда. Или в енум с состоянием

stvort commented on src/main/java/ru/otus/rzdtelegrambot/botapi/TelegramFacade.java:
```diff
@@ -0,0 +1,80 @@
+        BotState botState;
+        SendMessage replyMessage;
+
+        switch (inputMsg) {
+            case "Найти поезда":
+                botState = BotState.TRAINS_SEARCH;
+                break;
+            case "Мои подписки":
+                botState = BotState.SHOW_SUBSCRIPTIONS_MENU;
+                break;
+            case "Справочник станций":
+                botState = BotState.STATIONS_SEARCH;
+                break;
+            case "Помощь":
+                botState = BotState.SHOW_HELP_MENU;
+                break;
+            default:
+                botState = userDataCache.getUsersCurrentBotState(userId);
+                break;
+        }
```
Ну тут понятно. Хотя бы локализовать кнопки. Но лучше попробовать сделать еще гибче)

stvort commented on src/main/resources/application.properties:
```diff
@@ -1,12 +1,21 @@
+server.port=5000
```
Это не обязательно, но можно хранить в yml настройки бута)

stvort commented on src/main/resources/application.properties:
```diff
@@ -1,12 +1,21 @@
-#select proxy type: [HTTP | SOCKS4| SOCKS5 |NO_PROXY]
 telegrambot.userName=@RZDTicketsMonitoringBot
-telegrambot.token=
+telegrambot.botToken=1035774480:AAErNp8e7LlUvG0PuLL-SFSmDC96DGOjjQ0
```
Пропертя бута можно задавать через командную строку если тут задать плейсхолдер

viacheslaev committed:
Merge branch 'master' into develop
